execution/commitment/commitmentdb/commitment_context.go

```diff
@@ -442,18 +554,17 @@ type TrieContext struct {
 	putter kv.TemporalPutDel
 	txNum  uint64
 
-	limitReadAsOfTxNum uint64
-	stepSize           uint64
-	withHistory        bool // if true, do not use history reader and limit to domain files only
-	trace              bool
+	stepSize    uint64
+	trace       bool
+	stateReader StateReader
 }
 
 func (sdc *TrieContext) Branch(pref []byte) ([]byte, kv.Step, error) {
 	return sdc.readDomain(kv.CommitmentDomain, pref)
 }
 
 func (sdc *TrieContext) PutBranch(prefix []byte, data []byte, prevData []byte, prevStep kv.Step) error {
-	if sdc.limitReadAsOfTxNum > 0 && sdc.withHistory { // do not store branches if explicitly operate on history
+	if sdc.stateReader.HasFullHistory() { // do not store branches if explicitly operate on history
```

Comment:
`HasFullHistory` - seems wrong name of method? 
because: `sdc.limitReadAsOfTxNum > 0 && sdc.withHistory` look for me as "user asked to read from history?" while `HasFullHistory` looks for me as "does Erigon have enough data?"

Reply:
Strictly speaking, `sdc.limitReadAsOfTxNum > 0 && sdc.withHistory` seems more like "has full history" because in old code when `SetLimitReadAsOfTxNum` is called then `sdc.withHistory=!domainFiles`, so the real meaning of `withHistory` seems to be `withRecentFiles` and I read the condition above as `history is enabled and with recent files`, i.e. full.
Anyway, I agree that _probably_ the existing code intent is "user asked to read from history?", but I've preferred to reproduce exactly the same semantics, that's why I've chosen `HasFullHistory` (which is `true` for full history reader and `false` for history-without-recent-files reader).

Reply:
plz, help to include my 2 comments into stateReaders comments. to refer in future.

Reply:
ok, i will include this clarification into comments and also rename `HasFullHistory` -> `WithHistory`

Reply:
Regarding https://github.com/erigontech/erigon/pull/17413#discussion_r2425191309 :

why do we need to hide the recent files, when we can just use `GetAsOf()` ?